setup.py

```diff
@@ -37,6 +47,22 @@
 
 debug_flags = ['DEBUG_COSTLY_THROW']
 
+version_info = sys.version_info
+
+python_min_ver = (3, 4, 0)
+python_ver = (version_info.major, version_info.minor, version_info.micro)
+
+if python_ver < python_min_ver:
+    txt = 'Python version {0}.{1}.{2} ' \
+          'lower than the required version >= {3}.{4}.{5}.'
+
+    warnings.warn(txt.format(*(python_ver + python_min_ver)))
+
+if sys.platform == 'darwin':
+    from distutils import sysconfig
+    vars = sysconfig.get_config_vars()
+    vars['LDSHARED'] = vars['LDSHARED'].replace('-bundle', '-dynamiclib')
```

Comment:
Is this to avoid making a bundle? Maybe should it be commented explaining why we should avoid it